test/hip_device.cpp

```diff
@@ -77,21 +77,24 @@ namespace rocwmma
         mSharedMemSize = mProps.sharedMemPerBlock;
         mCuCount       = mProps.multiProcessorCount;
         mMaxFreqMhz    = static_cast<int>(static_cast<double>(mProps.clockRate) / 1000.0);
+        mCurFreqMhz = mMaxFreqMhz;
 
-        CHECK_RSMI_ERROR(rsmi_init(0));
+#ifdef ROCWMMA_BENCHMARK_TESTS
+        bool smiErrorFlag = false;
+        CHECK_RSMI_ERROR(rsmi_init(0), smiErrorFlag);
```

Comment:
If there are errors returned from smi at any time, we should not proceed with any values we try to obtain.
E.g. 
- If init fails, don't try any further smi calls
- If rsmi_num_monitor_devices fails, don't use the smiCount
- if rsmi_dev_pci_id_get fails, don't use rmsiPCIID
- The fallback at the end is if the error flag is set, use the max frequency reported by HipDevice

Reply:
In summary, should be additional checks on smiErrorFlag before proceeding to further smi calls